### PR TITLE
Change deployment for Linux premium to local RFP

### PR DESCRIPTION
**Pull request by ankitkumarr**
Fixes https://github.com/Azure/azure-functions-core-tools/issues/2502

cc: @anirudhgarg

### CODE REVIEW

File: src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs [hunk 1]
```diff
@@ -395,13 +395,14 @@ private async Task PublishFunctionApp(Site functionApp, GitIgnoreParser ignorePa
             else if (functionApp.IsLinux && functionApp.IsElasticPremium)
             {
                 // Elastic Premium Linux
-                shouldSyncTriggers = await HandleElasticPremiumLinuxPublish(functionApp, zipStreamFactory);
+                shouldSyncTriggers = true;
```
Comment (ankitkumarr) [inline src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs:398]: For premium, it's better to always sync trigger to notify the platform (remote build or not)

File: src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs [hunk 2]
```diff
@@ -469,23 +470,35 @@ private async Task SyncTriggers(Site functionApp)
...
+                && functionApp.AzureAppSettings.ContainsKey(Constants.WebsiteContentShared))
+            {
+                await HandleLinuxAppServicePublish(functionApp, zipStreamFactory);
+                return;
+            }
+
+            ColoredConsole.WriteLine(WarningColor($"Your Linux Premium function app does not have Azure Files App Settings" +
+                $" ('{Constants.WebsiteContentAzureFileConnectionString}', '{Constants.WebsiteContentShared}') configured. Azure Files setup is required to enable" +
+                $"persistent storage in Linux Premium function apps." +
+                $" For more info, see https://docs.microsoft.com/en-us/azure/azure-functions/functions-recover-storage-account#required-application-settings"));
```
Comment (ankitkumarr) [inline src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs:486]: I think we'll need a better aka.ms link here. @anirudhgarg or @amamounelsayed, do either of you know if we have some proper doc describing the need for Azure Files settings in Linux Premium that we could create aka.ms link for? I couldn't find one, but wanted to check first.

**Review by Hazhzeng — APPROVED** [review]

**Review by amamounelsayed — APPROVED** [review]

File: src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs [hunk 3]
```diff
@@ -474,23 +475,35 @@ private async Task SyncTriggers(Site functionApp)
...
+                await HandleLinuxAppServicePublish(functionApp, zipStreamFactory);
+                return;
+            }
+
+            ColoredConsole.WriteLine(WarningColor($"Your Linux Premium function app does not have Azure Files Application Settings" +
+                $" ('{Constants.WebsiteContentAzureFileConnectionString}', '{Constants.WebsiteContentShared}') configured. Azure Files setup is required to enable" +
+                $"persistent storage in Linux Premium function apps." +
+                $" For more info, see https://aka.ms/linux-ep-storage"));
+
+            // If Azure Files settings are not present, we need to do Run From Package = URL deployment
```
Comment (balag0) [inline src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs:493]: I thought local zip can work with AppService storage as well?

Comment (ankitkumarr) [reply src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs:493]: @balag0, in Premium scenarios with cross-stamp scaling in play, the storage in one instance may not be same as the storage in another instance of the same app (If not using Azure files, and relying on the stamp based storage). So, an instance in a different stamp may not be able to access the app content.
Using Run From Package = URL at least ensures that the apps continue to work in those case.

Comment (balag0) [reply src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs:493]: Yes, But if the app is not configured with AzureFiles it wont be eligible for cross stamp scale in the first place?

**Comment by kshyju** [comment]
@patelchandni Could you review this and see this change is valid/relevant?

**Comment by patelchandni** [comment]
Tagging @khkh-ms owner of deployment in core tools. Could you review this and see this change is valid/relevant?

**Comment by soninaren** [comment]
Closing the PR as it is against v3.x that is not released anymore.

**Pull request closed without merging**